src/configurator.rb

```diff
@@ -112,7 +110,11 @@ def initialize(identifier:, properties:, configure_file:,
       def ask(highline)
         ask_method = "ask_#{type}_question"
         self.send(ask_method, highline) do |highline_question|
-          highline_question.default = default
+          if !default.to_s.empty?
+            highline_question.default = default
+          elsif !optional
```

Comment:
I think `if !optional` would read clearer as `if required` - you could extract a `required` method which is just defined as `!optional`; or (possibly simpler) instead just save the original field as `@required = !properties[:optional]`, since we don't need to access `optional` anywhere other than here.